### PR TITLE
Add ConstantSourceNode

### PR DESCRIPTION
**Pull request by chrisdavidmills**

### CODE REVIEW

File: api/ConstantSourceNode.json [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+              "version_added": false
+            },
+            "chrome": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": false
+            },
+            "edge": {
+              "version_added": null
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: false

File: api/ConstantSourceNode.json [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+            },
+            "edge_mobile": {
+              "version_added": null
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: false

File: api/ConstantSourceNode.json [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+            },
+            "firefox": {
+              "version_added": "52"
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: I would add "version_removed": "53" with the note saying the property is still available but via the inheritance of AudioScheduledSourceNode.

File: api/ConstantSourceNode.json [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+            },
+            "firefox_android": {
+              "version_added": "52"
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: I would add "version_removed": "53" with the note saying the property is still available but via the inheritance of AudioScheduledSourceNode.

File: api/ConstantSourceNode.json [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+              "version_added": false
+            },
+            "chrome": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": false
+            },
+            "edge": {
+              "version_added": null
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: false

File: api/ConstantSourceNode.json [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+            },
+            "edge_mobile": {
+              "version_added": null
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: false

File: api/ConstantSourceNode.json [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+            },
+            "firefox": {
+              "version_added": false,
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: I would add "version_removed": "53" with the note saying the property is still available but via the inheritance of AudioScheduledSourceNode.

File: api/ConstantSourceNode.json [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+              "notes": "Firefox initially implemented this event handler directly on <code>ConstantSourceNode</code>, but now implements it on the parent <code>AudioScheduledSourceNode</code> interface as per spec."
+            },
+            "firefox_android": {
+              "version_added": false,
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: I would add "version_removed": "53" with the note saying the property is still available but via the inheritance of AudioScheduledSourceNode.

File: api/ConstantSourceNode.json [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+              "version_added": false
+            },
+            "chrome": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": false
+            },
+            "edge": {
+              "version_added": null
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: false

File: api/ConstantSourceNode.json [hunk 10]
```diff
@@ -0,0 +1,319 @@
...
+            },
+            "edge_mobile": {
+              "version_added": null
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: false

File: api/ConstantSourceNode.json [hunk 11]
```diff
@@ -0,0 +1,319 @@
...
+            },
+            "firefox": {
+              "version_added": false,
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: I would add "version_removed": "53" with the note saying the property is still available but via the inheritance of AudioScheduledSourceNode.

File: api/ConstantSourceNode.json [hunk 12]
```diff
@@ -0,0 +1,319 @@
...
+              "notes": "Firefox initially implemented this method directly on <code>ConstantSourceNode</code>, but now implements it on the parent <code>AudioScheduledSourceNode</code> interface as per spec."
+            },
+            "firefox_android": {
+              "version_added": false,
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: I would add "version_removed": "53" with the note saying the property is still available but via the inheritance of AudioScheduledSourceNode.

File: api/ConstantSourceNode.json [hunk 13]
```diff
@@ -0,0 +1,319 @@
...
+              "version_added": false
+            },
+            "chrome": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": false
+            },
+            "edge": {
+              "version_added": null
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: false

File: api/ConstantSourceNode.json [hunk 14]
```diff
@@ -0,0 +1,319 @@
...
+            },
+            "edge_mobile": {
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: false

File: api/ConstantSourceNode.json [hunk 15]
```diff
@@ -0,0 +1,319 @@
...
+              "version_added": null
+            },
+            "firefox": {
+              "version_added": false,
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: I would add "version_removed": "53" with the note saying the property is still available but via the inheritance of AudioScheduledSourceNode.

File: api/ConstantSourceNode.json [hunk 16]
```diff
@@ -0,0 +1,319 @@
...
+              "notes": "Firefox initially implemented this method directly on <code>ConstantSourceNode</code>, but now implements it on the parent <code>AudioScheduledSourceNode</code> interface as per spec."
+            },
+            "firefox_android": {
+              "version_added": false,
+              "notes": "Firefox initially implemented this method directly on <code>ConstantSourceNode</code>, but now implements it on the parent <code>AudioScheduledSourceNode</code> interface as per spec."
```
Comment (teoli2003) [inline api/ConstantSourceNode.json]: I would add "version_removed": "53" with the note saying the property is still available but via the inheritance of AudioScheduledSourceNode.

**Comment by chrisdavidmills** [comment]
I've made most of these changes. I don't think start should be unlisted here and just put on AudioScheduledSourceNode, as it was still available here at some point. People might come looking for it here.

**Comment by teoli2003** [comment]
Yes, of course for start (I think I changed my mind at some point, and forgot to remove the comment before finishing the review.

I'm seeing some:
"version_added": false,
 "version_removed": "53",

This should be possible (you can't remove something that has never been added. It should be version_added: true or a version number. I think you want "52" here:
"version_added": "52"
"version_removed": "53",
"notes": …

**Comment by chrisdavidmills** [comment]
I have made those changes and tried pushing them, but they are not appearing here, and git is saying that I am up-to-date with origin. I don't know what's happening here.

**Pull request merged**